### DISCUSSION TITLE
Specify custom name for `{service}-{stage}-custom-resource-apigw-cw-role` lambda function

### DISCUSSION THREAD

**Discussion by dan-bowen** [serverless/serverless#11305] [Q&A]
This resource is created by the framework but I don't see a way to give it a name that I want. The framework naming convention is preventing me from scoping down the permissions on it.

Is there a way to give this custom resource a custom name?

For context, my deployments are failing and I am going down a huge rabbit hole trying to make sure this resource has the permissions it needs. Being able to control the name of it will help me restrict permissions according to the same naming convention as the rest of my resources.

Is there a definitive resource somewhere that explains the purpose of these custom resources and how to manage the permissions for them?

**Comment by mattshine** [comment]
To hop onto this custom lambda function and its use case - is there a way to change the architecture of this function? Or control anything about it in any way, like the name, runtime, arch, etc?
